src/io.rs

```diff
@@ -429,22 +443,42 @@ pub enum SupplyChainStateReply {
     Participants(Participants),
     /// Should be returned from [`SupplyChainStateQuery::NFTProgram`].
     NFTProgram(ActorId),
+    /// Should be returned from [`SupplyChainStateQuery::ExistingItems`].
+    ExistingItems(BTreeMap<ItemId, ItemInfo>),
+    /// Should be returned from [`SupplyChainStateQuery::Roles`].
+    Roles(Roles),
+}
+
+bitflags::bitflags! {
+    /// Roles for interacting with a supply chain.
+    ///
+    /// For the descriptions of the first 3, see the documentation of
+    /// [`Participants`].
+    ///
+    /// Can be queried by [`SupplyChainStateQuery::Roles`].
+    #[derive(Encode, Decode, TypeInfo)]
+    pub struct Roles: u8 {
```

Comment:
Isn't Enum type better fits for `Roles`?

Reply:
Not in this case, I think. Bitflags are much easier to process and store between a frontend and backend than enums. And they also reduce a size of a contract.